python2verilog/frontend/generator.py

```diff
@@ -377,27 +411,24 @@ def __parse_yield(self, node: pyast.Yield, prefix: str):
         """
         yield <value>;
         """
-
-        match node.value:
-            case pyast.Tuple(elts=elts):
-                stmts = [self.__parse_expression(c) for c in elts]
-            case pyast.expr():
-                stmts = [self.__parse_expression(node.value)]
-            case _:
-                raise TypeError(f"Expected tuple {type(node.value)} {pyast.dump(node)}")
+        if isinstance(node.value, pyast.Tuple):
+            stmts = [self.__parse_expression(c) for c in node.value.elts]
+        elif isinstance(node.value, pyast.expr):
+            stmts = [self.__parse_expression(node.value)]
+        else:
+            raise TypeError(f"Expected tuple {type(node.value)} {pyast.dump(node)}")
```

Comment:
If you want to use `match` statements, you can always write a class to achieve the same functionality.

E.g. (you will have to rewrite this in proper python)
```python
class Switch():
    # constructor
    def Switch() {
        self.callbacks = {}
        self.default = lambda _: return # some default function

    # Handle case events
    def case(key, func):
        self.callbacks[key] = func
        return self
 
    # default case
    def default(func):
        self.default = func
        return self

    # Overwrite the call method so that you can invoke this class as a function
    # var will be the variable you're switching on
    def __call__(var):
        return self.callbacks[key]() if key in self.callbacks else self.default()
```

and then you can create the match as so:

```python
new Switch()
.case(pyast.Tuple(elts=elts), lambda: stmts = [self.__parse_expression(c) for c in elts])
.case(pyast.expr(), lambda: [self.__parse_expression(node.value)])
.case(_, lambda: raise TypeError(f"Expected tuple {type(node.value)} {pyast.dump(node)}"))
(node.value)

```

You can also extend the functionality so that you can pass in arguments to the callback functions